cmd/fleetctl/get_test.go

```diff
@@ -1053,6 +1053,185 @@ spec:
 	assert.Equal(t, expectedJson, runAppForTest(t, []string{"get", "query", "--json", "query1"}))
 }
 
+// TestGetQueriesAsObservers tests that when observers run `fleectl get queries` they
+// only get queries that they can execute.
+func TestGetQueriesAsObserver(t *testing.T) {
+	_, ds := runServerWithMockedDS(t)
+
+	setCurrentUserSession := func(user *fleet.User) {
+		user, err := ds.NewUser(context.Background(), user)
+		require.NoError(t, err)
+		ds.SessionByKeyFunc = func(ctx context.Context, key string) (*fleet.Session, error) {
+			return &fleet.Session{
+				CreateTimestamp: fleet.CreateTimestamp{CreatedAt: time.Now()},
+				ID:              1,
+				AccessedAt:      time.Now(),
+				UserID:          user.ID,
+				Key:             key,
+			}, nil
+		}
+		ds.UserByIDFunc = func(ctx context.Context, id uint) (*fleet.User, error) {
+			return user, nil
+		}
+	}
+
+	ds.ListQueriesFunc = func(ctx context.Context, opt fleet.ListQueryOptions) ([]*fleet.Query, error) {
+		return []*fleet.Query{
+			{
+				ID:             42,
+				Name:           "query1",
+				Description:    "some desc",
+				Query:          "select 1;",
+				ObserverCanRun: false,
+			},
+			{
+				ID:             43,
+				Name:           "query2",
+				Description:    "some desc 2",
+				Query:          "select 2;",
+				ObserverCanRun: true,
+			},
+			{
+				ID:             44,
+				Name:           "query3",
+				Description:    "some desc 3",
+				Query:          "select 3;",
+				ObserverCanRun: false,
+			},
+		}, nil
+	}
+
+	for _, tc := range []struct {
+		name string
+		user *fleet.User
+	}{
+		{
+			name: "global observer",
+			user: &fleet.User{
+				ID:         1,
+				Name:       "Global observer",
+				Password:   []byte("p4ssw0rd.123"),
+				Email:      "go@example.com",
+				GlobalRole: ptr.String(fleet.RoleObserverPlus),
+			},
+		},
+		{
+			name: "team observer",
+			user: &fleet.User{
+				ID:         2,
+				Name:       "Team observer",
+				Password:   []byte("p4ssw0rd.123"),
+				Email:      "tm@example.com",
+				GlobalRole: nil,
+				Teams:      []fleet.UserTeam{{Role: fleet.RoleObserver}},
+			},
+		},
+		{
+			name: "observer of multiple teams",
+			user: &fleet.User{
```

Comment:
Could you please also add another test case for when the user is not an observer/observer+ on all teams? Just to cover all cases

Reply:
Good idea. Done fbb8d8b56.